/*
      Dopo aver modificato il .proto file:
          python3 -m grpc_tools.protoc -I../../protos --python_out=. --grpc_python_out=. ../../protos/managepassword.proto

      Questo genera
          - savepwd_pb2.py che contiene le classi di richiesta e risposta
          - savepwd_pb2_grpc.py che contiene le classi client e server
*/

syntax = "proto3";

option java_multiple_files = true;
option java_package = "/SDCC/services";
option java_outer_classname = "SavePwProto";
option objc_class_prefix = "SPW";
package managepw;

// The  service definition.
service Saver {

  rpc SavePassword (SavePasswordRequest) returns (SavePasswordReply) {}
  rpc doList (ListPasswordRequest) returns (ListPasswordReply) {}

}

message SavePasswordRequest {
  string email = 1;
  string password = 2;
  string service = 3;
}

message SavePasswordReply {
  bool isStored = 1;
}

message ListPasswordRequest {
  string email = 1;
}

message ListPasswordReply {
  map<string, string> list = 1;
}